mirgecom/transport.py

```diff
@@ -0,0 +1,222 @@
+($d_{\alpha}$).
+
+.. autoclass:: TransportDependentVars
+.. autoclass:: TransportModel
+.. autoclass:: SimpleTransport
+.. autoclass:: PowerLawTransport
+"""
+
+__copyright__ = """
+Copyright (C) 2021 University of Illinois Board of Trustees
+"""
+
+__license__ = """
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+"""
+
+from dataclasses import dataclass
+import numpy as np
+from meshmode.mesh import BTAG_ALL, BTAG_NONE  # noqa
+from mirgecom.fluid import ConservedVars
+from mirgecom.eos import GasEOS
+
+
+@dataclass
+class TransportDependentVars:
+    """State-dependent quantities for :class:`TransportModel`.
+
+    Prefer individual methods for model use, use this
+    structure for visualization or probing.
+
+    .. attribute:: bulk_viscosity
+    .. attribute:: viscosity
+    .. attribute:: thermal_conductivity
+    .. attribute:: species_diffusivity
+    """
+
+    bulk_viscosity: np.ndarray
+    viscosity: np.ndarray
+    thermal_conductivity: np.ndarray
+    species_diffusivity: np.ndarray
+
+
+class TransportModel:
+    r"""Abstract interface to thermo-diffusive transport model class.
+
+    Transport model classes are responsible for
+    computing relations between fluid or gas state variables and
+    thermo-diffusive transport properties for those fluids.
+
+    .. automethod:: bulk_viscosity
+    .. automethod:: viscosity
+    .. automethod:: thermal_conductivity
+    .. automethod:: species_diffusivity
+    .. automethod:: viscosity2
+    """
+
+    def bulk_viscosity(self, eos: GasEOS, cv: ConservedVars):
+        r"""Get the bulk viscosity for the gas (${\mu}_{B}$)."""
+        raise NotImplementedError()
+
+    def viscosity(self, eos: GasEOS, cv: ConservedVars):
+        r"""Get the gas dynamic viscosity, $\mu$."""
+        raise NotImplementedError()
+
+    def viscosity2(self, eos: GasEOS, cv: ConservedVars):
+        r"""Get the 2nd coefficent of viscosity, $\lambda$."""
+        raise NotImplementedError()
```

Comment:
`viscosity` and `viscosity2` are not very helpful function names. Why not just call them bulk and dynamic viscosities (or something else)? That to me seems way more helpful.